src/extension/settings/configReaders.ts

```diff
@@ -108,12 +110,21 @@ export function readAndValidateUserModelsParams(
                 jsonSchemaValidator
             )
         );
+    const deepSeekUserParams: DeepSeekUserModelParams[] =
+        config.deepSeekModelsParameters.map((params: any) =>
+            validateAndParseJson(
+                params,
+                deepSeekUserModelParamsSchema,
+                jsonSchemaValidator
+            )
+        );
 
     validateIdsAreUnique([
         ...predefinedProofsUserParams,
         ...openAiUserParams,
         ...grazieUserParams,
         ...lmStudioUserParams,
+        ...deepSeekUserParams,
     ]);
     validateApiKeysAreProvided(openAiUserParams, grazieUserParams);
```

Comment:
You forgot to connect `DeepSeek` here - this check will correctly handle the initial `None` value of the API key.